### ISSUE TITLE
Terraform shows diff in start_timestamp of Snowflake's resource monitor  just after apply

### ISSUE THREAD

**Issue by LucasMMota** [Snowflake-Labs/terraform-provider-snowflake#1821]
**Provider Version**

provider registry.terraform.io/snowflake-labs/snowflake v0.63.0

**Terraform Version**

Terraform v1.4.5

**Describe the bug**

The start date of resource monitor cannot be set as saved in the database (e.g.: "2023-04-24T03:00:00Z"). Although if we use the expected format by the provider, the plan presents a difference just after the apply.

**Expected behavior**

No diff after the apply. Or provider accepts the format stored in DB with the timezone (e.g.: "2023-04-24T03:00:00Z")

**Code samples and commands**

The resource monitor object:

```
 {
    name            = "RM_WH_REPORTING"
    credit_quota    = 30
    frequency       = "DAILY"
    start_timestamp = "2023-05-23"
    notify_triggers = [80, 90]
    suspend_immediate_trigger = 100
    notify_users    = ["SNOWFLAKE_ALERTS_SLACK_INTEGRATION"]
    grant_privileges_by_role = {
      TRANSFORMER : ["MONITOR"]
    }
  }
```

The plan:

```terraform
 # module.resource_monitor["RM_WH_REPORTING"].snowflake_resource_monitor.monitor will be created
  + resource "snowflake_resource_monitor" "monitor" {
      + credit_quota              = 30
      + frequency                 = "DAILY"
      + id                        = (known after apply)
      + name                      = "RM_WH_REPORTING"
      + notify_triggers           = [
          + 80,
          + 90,
        ]
      + notify_users              = [
          + "SNOWFLAKE_ALERTS_SLACK_INTEGRATION",
        ]
      + set_for_account           = false
      + start_timestamp           = "2023-05-23"
      + suspend_immediate_trigger = 100
    }
```

The terraform plan **after applying**:

```terraform
  # module.resource_monitor["RM_WH_REPORTING"].snowflake_resource_monitor.monitor will be updated in-place
  ~ resource "snowflake_resource_monitor" "monitor" {
        id                        = "RM_WH_REPORTING"
        name                      = "RM_WH_REPORTING"
      ~ start_timestamp           = "2023-05-23T00:00:00Z" -> "2023-05-23"
```

**Comment by LucasMMota** [comment]
also if you try to not use state time (to use the immediately option) this error will throw:
`error creating resource monitor RM_WH_REPORTING1 err = 090259 (42601): Must specify frequency and start time together. `

**Comment by sfc-gh-jcieslak** [comment]
Hello 👋
Please refer to this response - https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1175#issuecomment-2331300488. I'll let you know whenever the updated resource is available.

**Comment by sfc-gh-jcieslak** [comment]
Hey 👋
The new and refactored resource monitor was released yesterday in version 0.96.0 of the provider. Please migrate with [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0950--v0960) and let me know if the issue has been resolved so we could close the ticket. Thanks in advance 🙏.

**Comment by sfc-gh-jcieslak** [comment]
Closing due to long inactivity. Please, create another issue if you think the problem is still not resolved.
